pennylane/measurements.py

```diff
@@ -740,6 +742,34 @@ def circuit(x):
     >>> circuit(0.5)
     {'00': 3, '01': 1}
 
+    Per default, outcomes that were not observed will not be included in the dictionary.
+    Passing all_outcomes=True will create a dictionary that displays all possible outcomes.
+
+    .. code-block:: python3
+
+    dev = qml.device("default.qubit", wires=2, shots=4)
+
+    @qml.qnode(dev)
+    def circuit():
+        qml.PauliX(wires=0)
+        return qml.counts()
+
+    Executing this QNode shows only the observed state:
+
+    >>> circuit()
+    {'01': 4}
+
```

Comment:
Might be good to move the note
>     Passing all_outcomes=True will create a dictionary that displays all possible outcomes.

Just above the other example.
```suggestion

    .. code-block:: python3

    dev = qml.device("default.qubit", wires=2, shots=4)

    @qml.qnode(dev)
    def circuit():
        qml.PauliX(wires=0)
        return qml.counts()

    Executing this QNode shows only the observed state:

    >>> circuit()
    {'01': 4}

    Passing all_outcomes=True will create a dictionary that displays all possible outcomes.

```